frontend/public/components/secrets/create-secret.tsx

```diff
@@ -144,6 +163,308 @@ const withSecretForm = (SubForm) => class SecretFormComponent extends React.Comp
+  }
+  newImageSecretEntry() {
+    return {
+      address: '',
+      username: '',
+      password: '',
+      email: '',
+      auth: '',
+    };
+  }
+  imageSecretObjectToArray(imageSecretObject) {
+    const imageSecretArray = [];
+    if (_.isEmpty(imageSecretObject)) {
+      return _.concat(imageSecretArray, this.newImageSecretEntry());
+    }
+    _.each(imageSecretObject, (v, k) => {
+      // Decode and parse 'auth' in case 'username' and 'password' are not part of the secret.
+      const decodedAuth = window.atob(_.get(v, 'auth', ''));
+      const parsedAuth = _.isEmpty(decodedAuth) ? _.fill(Array(2), '') : _.split(decodedAuth, ':');
+      imageSecretArray.push({
+        address: k,
+        username: _.get(v, 'username', parsedAuth[0]),
+        password: _.get(v, 'password', parsedAuth[1]),
+        email: _.get(v, 'email', ''),
+        auth: _.get(v, 'auth', ''),
+      });
+    });
+    return imageSecretArray;
+  }
+  imageSecretArrayToObject(imageSecretArray) {
+    const imageSecretsObject = {};
+    _.each(imageSecretArray, (entry) => {
+      const entryCredentials = {
+        username: entry.username,
+        password: entry.password,
+        auth: entry.auth,
+        email: entry.email
+      };
+      imageSecretsObject[entry.address] = entryCredentials;
+    });
+    return imageSecretsObject;
+  }
+  // Image secrets entry address can't be duplicate in the secret, else the first one will be deleted.
+  checkEntryAddressForDuplicates(updatedEntryData, entryID) {
+    let hasDuplicate = false;
+    _.each(this.state.secretEntriesArray, (entry, index) => {
+      // Don't check the same entry, continue.
+      if (entryID === index) {
+        return;
+      }
+      if (_.isEqual(entry.address, updatedEntryData.address)) {
+        hasDuplicate = true;
+        return false;
+      }
+    });
+    return hasDuplicate;
+  }
+  onDataChanged (updatedEntryData, entryID) {
+    const secretEntriesArray = this.state.secretEntriesArray;
+    secretEntriesArray.splice(entryID, 1, updatedEntryData);
+    this.setState({
+      secretEntriesArray: secretEntriesArray,
+      hasDuplicate: this.checkEntryAddressForDuplicates(updatedEntryData, entryID)
+    }, () => {
+      const imageSecretObject = this.imageSecretArrayToObject(this.state.secretEntriesArray);
+      this.props.onChange(this.state.isDockerconfigjson ? {[AUTHS_KEY]: imageSecretObject} : imageSecretObject);
+    });
+  }
+  render() {
+    const secretEntriesList = [];
+    _.each(this.state.secretEntriesArray, (entry, index) => {
+      secretEntriesList.push(<ConfigEntryForm key={index} id={index} entry={entry} onChange={this.onDataChanged} />);
+    });
+    return (
+      <React.Fragment>
+        {secretEntriesList}
+        { this.state.hasDuplicate && <div className="co-create-secret-warning">Updated <b>Registry Server Address</b> is duplicate. Remove it, or it will be removed upon saving.</div> }
```

Comment:
Yeah, I wasn't sure if we want to check for the dup. here, since it doesn't make sense to have different credentials pointing to the same address in one secret. 
Other think is if the users know that :-/